src/autogpt_plugins/email/email_plugin/test_email_plugin.py

```diff
@@ -561,5 +571,52 @@ def test_clean_email_body(self, mock_imap):
         mock_imap.return_value.search.assert_called_once_with(None, "UNSEEN")
         mock_imap.return_value.fetch.assert_called_once_with(b"1", "(BODY.PEEK[])")
 
+    # Test for truncating emails
```

Comment:
We should also write some unit tests that test the various settings e.g. max_num_emails:

Here some suggestions
```
    @patch('os.getenv', return_value='1000')
    def test_empty_list(self, mock_getenv):
        truncator = EmailTruncator()
        messages = []
        self.assertEqual(truncator.truncate_emails(messages), [])
    
    @patch('os.getenv')
    def test_max_characters_breached(self, mock_getenv):
        truncator = EmailTruncator()
        mock_getenv.side_effect = ['10', '10', '100']
        messages = [{"Message Body": "Hello world!"}]
        truncated_messages = truncator.truncate_emails(messages)
        self.assertTrue(len(str(truncated_messages)) <= 10)
    
    @patch('os.getenv')
    def test_max_num_emails_breached(self, mock_getenv):
        truncator = EmailTruncator()
        mock_getenv.side_effect = ['1000', '1', '100']
        messages = [{"Message Body": "Hello world!"}, {"Message Body": "Goodbye world!"}]
        truncated_messages = truncator.truncate_emails(messages)
        self.assertEqual(len(truncated_messages), 1)
    
    @patch('os.getenv')
    def test_max_email_characters_breached(self, mock_getenv):
        truncator = EmailTruncator()
        mock_getenv.side_effect = ['1000', '10', '10']
        messages = [{"Message Body": "Hello world!"}]
        truncated_messages = truncator.truncate_emails(messages)
        self.assertTrue(len(str(truncated_messages[0]["Message Body"])) <= 10)
```